### ISSUE TITLE
amp-inline-gallery/amp-lightbox-gallery: Non-text content has insufficient color contrast

### ISSUE THREAD

**Issue by TetraLogicalHelpdesk** [ampproject/amphtml#31157]
## WCAG Level
Level AA

## Priority
Low

## Type
Color

## Pages/screens/components affected
* [amp-inline-gallery](https://amp.dev/documentation/components/amp-inline-gallery/)
* [amp-lightbox-gallery](https://amp.dev/documentation/components/amp-lightbox-gallery/)

## Description
An important design consideration is to make sure there is enough contrast between the color of meaningful non-text elements (such as controls denoted purely through graphical elements, icons used as the only visible indicator for errors, charts/graphs, meaningful borders/outlines used to denote a selected item) and their background. Good color contrast ensures that all sighted users, including those who have a condition that affects their ability to perceive color, are able to see/understand meaningful non-text content.

The "dots" for the currently inactive slides in `amp-inline-gallery` and `amp-lightbox-gallery` (which use the `amp-carousel` component) use a semi-transparent grey color on a semi-transparent black background. Depending on the slide background, these dots have a contrast ratio below 3:1.

![Carousel dots against a bright slide - with the inactive dots showing insufficient contrast](https://user-images.githubusercontent.com/54933393/99181175-ae35cf00-2724-11eb-8d23-005bbc525c3d.png)

## User impact
Without sufficient contrast, content can be difficult to perceive and therefore difficult to identify. In extreme cases, content with low contrast may not be visible at all to people with color perception impairments.

In this case, users with impaired color perception may be unable to clearly perceive the number of slides present in the carousel.

## Required solution
Make sure that non-text content has a color contrast ratio of at least 3:1 against its background and the content that is adjacent to it.

Elements that are disabled or inactive, and elements that use the browser's default styles, are exempt.

This solution must be applied to all instances of the issue identified within the test sample, then applied to all other instances of the same issue identified throughout the rest of the website.

## Implementation guidance
In the case of the carousel dots, currently their color is set to a 35% transparent white color.

```
amp-inline-gallery-pagination[inset]
.i-amphtml-inline-gallery-pagination-dot {
  background-color: hsla(0,0%,100%,0.35);
}
```

The pagination element's background is set to a semi-transparent gray color.

```
amp-inline-gallery-pagination[inset]
.i-amphtml-inline-gallery-pagination-backdrop {
  ...
  opacity: 0.5;
}
```

We would recommend either darkening the background (for instance, making it less transparent by increasing its `opacity`), or increasing the opacity of the inactive pagination dots (changing the alpha of the `hsla` background colour), in order to achieve a contrast ratio of 3:1.

Note that doing this may reduce the visible difference between the currently active dot and the inactive dots. Ideally, their difference should also be very visually clear. We would therefore further recommend differentiating active and inactive dots not just by a difference in opacity, but through an additional visual indicator - for instance, by providing an additional border/outline to the active dots, or changing its appearance more clearly.

![Modified carousel dots - inactive dots with higher contrast, and active dot shown as an unfilled white circle](https://user-images.githubusercontent.com/54933393/99181182-b7bf3700-2724-11eb-97ac-a38683d5eec9.png)

## Test procedure(s)
Use these steps to confirm that the solution has been correctly applied to issues identified within the test sample, and to test the rest of the website for instances of the same issue:
1. Check the contrast of all non-text content using a tool such as the [TPG: Colour Contrast Analyser](https://developer.paciellogroup.com/resources/contrastanalyser/).
2. Check that the contrast level between the element and its adjacent content is at least 3:1.

### Definition of done
Complete all of these tasks before closing this issue or indicating it is ready for retest:
* All issues identified within the test sample have been resolved.
* The rest of the website  has been tested for the same issue.
* All issues identified throughout the rest of the website  have been resolved or filed as new issues.

## Related standards
* [WCAG 2.1 Success Criterion 1.4.11 Non-text Contrast (Level AA)](https://www.w3.org/TR/WCAG21/#non-text-contrast)

## More information
* [TPG: Colour Contrast Analyser](https://developer.paciellogroup.com/resources/contrastanalyser/)
* [WebAIM: Contrast and Color Accessibility - 1.4.11 Non-text Contrast](https://webaim.org/articles/contrast/#sc1411)
* [WebAIM: Color blindness](https://webaim.org/articles/visual/colorblind)

## Test data
Test date: November 2020
